runtime/src/chplcast.c

```diff
@@ -51,7 +51,7 @@ static int scanningNCounts(void) {
 }
 
 static int illegalFirstUnsChar(char c) {
-  return ((c < '0' || c > '9') && (c != '+'));
+  return ((c < '0' || c > '9') && (c != '+') && (c < 'a' || c > 'f') &&(c < 'A' || c > 'F'));
```

Comment:
I think this should just `return c == '-'`. I think trying to check against the other values is making it unnecessarily complicated. After all, we don't seem to need this level of checking in the signed case.

Reply:
Okay, I'm changing it to return  `c ==  '-'`